### ISSUE TITLE
Missing letter from name

### ISSUE THREAD

**Issue by koopermark** [jefffhaynes/XBee#55]
Hi, 

I dont know if this is an issue or if I am doing something wrong.

I am using win iot core UWP on the raspberry pi with Xbee S2C connected to USB.
The other Xbee S2C is connected to a PC using XCTU.

The Xbee connected to the PC has a NI (network identifier), when the following is run I expected to see the NI - it does but always misses the first letter. 

Example 
NI Set - Hello
name received - ello

    controller.NodeDiscovered += async (sender, args) =>
            {
                Debug.WriteLine("Connected Xbee Name    : " + args.Name);
            }

**Comment by jefffhaynes** [comment]
This could have something to do with the super temperamental UARTs on the raspberry pi. I’m not sure I ever got that completely working.  Serial support in UWP is bad in general and doubly so on the latest pis because apparently they play some tricks to free up other resources internally.

I’ll try to find time to look at it this weekend and see if there have been any advances I can leverage there.

Thanks,

Jeff


________________________________

**Comment by koopermark** [comment]
Thank you, this is my first venture into the UWP and with the PI, it has been a bit challenging to get the Xbee working in the first place - that was until I found your hard work - thank you

**Comment by jefffhaynes** [comment]
Yeah UWP/Pi/Core are all tantalizingly close to being really good but all seem to be lacking in some critical area.


________________________________

**Comment by koopermark** [comment]
Hi, I didnt want to start an issue with this on as it me not knowing what I am doing, but I was hoping you could help.

I am trying to asynchronously keep checking for any new Xbee's that can join the network but I cant figure it out, could you point me in the right direction.

I am looking at a small project that has Xbees that can join and disappear all the time, there will be one connected to the Pi which will always be on - but others will be turned on and off (not sleeping) which need to connect and send data to the Pi when they are powered up.

I know your not tech support and this library was shared out of your kindness, I would appreciate and pointers you have.
